### ISSUE TITLE
As a user, I want to have read-only access to all registry indexes

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/registry#350]
### Checked for duplicates

Yes - I've already checked

### 🧑‍🔬 User Persona(s)

Node Operator

### 💪 Motivation

...so that I can run tools against the read-only registry indexes. e.g. run validate-refs against all data in the database

### Acceptance Criteria

**Given** one lidvid that exists in the geo-registry index and one lidvid that exists in the naif-registry index
**When I perform** a query for each of those lidvids against those registries
**Then I expect** to be able to retrieve results

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

There are two ways to do this with harvest, mgr tools, and validate-refs (they all use registry-common):
1. make an index in AOSS that maps to all the other indices (may not be possible)
2. make index in the configuration either '.' or ';' or ':' or mix of any separate list instead of a single index. The Java SDK v2 supports assigning multiple indices instead of one. When multiple, make sure operation is read only.

**Comment by jordanpadams** [comment]
@al-niessner does the Java SDK support wildcards like `*`? This is something we can do in Kibana to search across indexes.

**Comment by al-niessner** [comment]
No. It takes a list. We can convert wildcards to list since the full list
is known as priori.

**Comment by sjoshi-jpl** [comment]
@jordanpadams @viviant100 @tloubrieu-jpl we will need someone from one of the nodes to confirm there Read access to all indexes. The reason we cannot test with users in EN group is because these users are already have read access to all indexes. Who can I reach out to for testing?

**Comment by tloubrieu-jpl** [comment]
Hi @sjoshi-jpl , I can volunteer tomorrow to be removed from EN node and be associated to a discipline node.

**Comment by sjoshi-jpl** [comment]
@tloubrieu-jpl I created a test user, we can test with it tomorrow.

**Comment by jordanpadams** [comment]
@sjoshi-jpl per [this discussion thread on validate](https://github.com/NASA-PDS/validate/issues/621#issuecomment-2591200718), instead of having a read-only role for all indices, could we just create a read-only alias index that maps to all the indices?

**Comment by jordanpadams** [comment]
e.g. 
```
POST _aliases
{
  "actions": [
    {
      "add": {
        "index": "en-registry",
        "alias": "readall-registry"
      }
    },
    {
      "add": {
        "index": "atm-registry",
        "alias": "readall-registry"
      }
    }
    // ...
  ]
}
```

**Comment by tloubrieu-jpl** [comment]
Sorry I missed the comment before, but as a first test result, with test user `testuser1` belonging to group `GEO`, created by @sjoshi-jpl:
- we can read all the indexes
- we can write in the geo index:

```
pds-registry-client '/geo-registry/_create/testttt' --data @/Users/loubrieu/tmp/test_doc.json --pretty
Response returned HTTP201 Created: b'{"_index":"geo-registry","_id":"testttt","_version":1,"result":"created","_shards":{"total":0,"successful":0,"failed":0},"_seq_no":0,"_primary_term":0}'
```



- we cannot write in the en index

```
% pds-registry-client '/en-registry/_create/testttt' --data @/Users/loubrieu/tmp/test_doc.json --pretty
Response returned HTTP403 Forbidden: b'{"error":{"root_cause":[{"type":"security_exception","reason":"Authorization failure for the following indices: [index/pds-mcp-registry-prod/en-registry]"}],"type":"security_exception","reason":"Authorization failure for the following indices: [index/pds-mcp-registry-prod/en-registry]"},"status":403}'
```

**Comment by tloubrieu-jpl** [comment]
@sjoshi-jpl the initial scope of the ticket is validated. Can you look at the new request of Jordan, regarding having a virtual index aggregating the others ?

We would also need to document how the authorization is managed, in the wiki or in terraform.

**Comment by jordanpadams** [comment]
@sjoshi-jpl @tloubrieu-jpl I am OK with not pursuing this alias index. It seems like a more concise approach for handling the authorization, but if it is not, then we can continue with the route @tloubrieu-jpl just tested.

**Comment by jordanpadams** [comment]
This is done through IAM.

@sjoshi-jpl has this been documented anywhere?

**Comment by tloubrieu-jpl** [comment]
This is documented on this page "User+Authorizations+for+OpenSearch+Serverless" on the internal wiki.

**Comment by tloubrieu-jpl** [comment]
Hi @c-suh ,

The lidvid is not required. You can search on sbnumd and en in the dev instance, and you should be able to see products in the 2 indices (en-registry, sbnumd-registry). You don't need search parameters (single_product.json).

You could also check the read-only by trying to write in sbnumd as an EN person. You can use harvest to do that (updating the node in the connection configuration file) or as you do with pds-registry-client using this type of command, https://docs.opensearch.org/docs/latest/api-reference/document-apis/update-document/

There are limitations in registry-client which would not allow you to create a document with a PUT HTTP operation.
